### ISSUE TITLE
Filesystem dock and import invisiable?

### ISSUE THREAD

**Issue by Jackerty** [godotengine/godot#87754]
### Tested versions

- Reproducible: commit 17e7f85c06366b427e5068c5b3e2940e27ff5f1d
- Not reproducible: The Archlinux Godot package (4.2.1-1) has previous buggy behaviour where moving the scene tree dock fixes the issue.

### System information

Arch linux [6.6.10.arch1-1] - AMD Ryzen 7 2700X Eight-Core Processor - GeForce RTX 2060 Rev. A

### Issue description

I have had, for a long time now, an issue where the filesystem dock and import dock are hidden "behind" the scene tree dock initially. I usually fixed this by moving the scene tree dock to another dock. That fix has now stopped working.

Here is a picture of what I see.
![FileSystemMissing](https://github.com/godotengine/godot/assets/2855916/c1066c88-5866-4991-8ac4-ee1375dc4fa9)

In the dock change popup, I do see that they should exist.
![Dock pop-up](https://github.com/godotengine/godot/assets/2855916/80bd1653-7ab3-491b-875f-a5d946e92c0f)

I can make them visible by adding debug code that sets the visibility of `dock_slot`s to true at the end of `EditorDockManager::load_docks_from_config`. They are gone if I move anything to a different dock.
![Screenshot_20240130_203921](https://github.com/godotengine/godot/assets/2855916/17bf628b-3c9e-492e-86d5-ef909836436b)

There is some graphical weirdness, so my hypothesis is something is wrong with choosing what docks are visible/layouting.

### Steps to reproduce

Compiling and then opening Godot on Archlinux. Dunno this may be a bug too specific for me and only me.

### Minimal reproduction project (MRP)

N/A

**Comment by Jackerty** [comment]
Tested with Archlinux godot package.

**Comment by Calinou** [comment]
Try using **Editor > Layouts > Default** to restore the default layout.

If all else fails, start the editor in [self-contained mode](https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html#self-contained-mode) on a new project so it uses default settings.

**Comment by Jackerty** [comment]
I have always been using default layout. I will checkout self-contained mode.

What I have debugged now at default layout `update_dock_slots_visibility` local variable `first_tab_visible` is following per dock slot. Slot 3  seem to be empty which does not sound right considering filesystem dock should be there.

```
Dock  0 : `first_tab_visible` =  -1
Dock  1 : `first_tab_visible` =  -1
Dock  2 : `first_tab_visible` =  0
Dock  3 : `first_tab_visible` =  -1
Dock  4 : `first_tab_visible` =  0
Dock  5 : `first_tab_visible` =  -1
Dock  6 : `first_tab_visible` =  -1
Dock  7 : `first_tab_visible` =  -1
```

**Comment by Jackerty** [comment]
I replaced global "editor_settings-4.tres" with one in the self contained mode and that does fix the issue.

**Comment by Jackerty** [comment]
Bug is caused by following line in the "editor_settings-4.tres":

```
_default_feature_profile = "Test"
```

It seems if the value is anything else (even fully in lower case) bug does not happen.

**Comment by AThousandShips** [comment]
For support questions and help please first turn to the other [community channels](https://godotengine.org/community), this is for bug reporting, not a help desk. Please use other channels *first* to see if this is a bug or user error 🙂

**Comment by Jackerty** [comment]
I didn't make a support question. What are you talking about?

**Comment by AThousandShips** [comment]
You were making a user error 🙂 that's why it's best to check with the community first, this wasn't a bug so just giving a friendly suggestion to check for that

**Comment by Jackerty** [comment]
I do not agree with you. I made a bug report. Yes, in the end, the fix was a user error, but there is still a bug at the bottom.

The Archlinux Godot package (4.2.1-1) has a bug where FileSystemDock can be made to reappear after it is disabled. I have seen this bug so many times it led me to have the wrong impression about my Godot installation. So when it was fixed, Godot's behaviour appeared to me very much a bug that got worse.

There is nothing in the guidelines recommending going to the community before a bug report. So it appears to me as a bug I make a bug report. How that bug report ends up being resolved does not matter.

I thank you for doing the bugsquad work. I do not find what you said friendly. I find it very snide.

**Comment by akien-mga** [comment]
I know for sure AThousandShips didn't mean to be disparaging. Written communication makes it hard to convey nuance, but there is no side in her suggestion.

Now, in this specific case I agree that it was reasonable to think this is a bug and report it as such. You're not the first user to be confused by having enabled some feature profile by mistake. So that's something we'll have to improve on the UX.

**Comment by AThousandShips** [comment]
I'm sorry that what I said came across as such, but I fail to see how using a clearly friendly smiley isn't defusing any assumption of snide or rudeness, I try my best to communicate in a difficult medium and I'd ask that you communicate back and take people at good faith, as the [code of conduct asks](https://godotengine.org/code-of-conduct/) 🙂

My point was that it's best to discuss things in community channels first, for many reasons, there's nothing wrong in reporting a bug here, but chances are you get better immediate feedback by that means, there's many reports popping up and can be hard to get someone to react and notice

But again, first assume good intent, and if you *do* feel that someone is behaving bad, communicate that to them first, instead of assuming intent where none was present, that's the best way to resolve situations, be considerate of the way written communication is hard to process (but do make an effort and take into account obvious signals)

**Comment by AThousandShips** [comment]
I mean snide is not something separate from intention, so I simply wasn't snide, so I'd appreciate a proper apology, but it doesn't really matter
